tests/cases/compiler/jsxElementType.tsx

```diff
@@ -0,0 +1,73 @@
+// @strict: true
+// @jsx: react
+/// <reference path="/.lib/react16.d.ts" />
+import * as React from "react";
+
+type React18ReactFragment = ReadonlyArray<React18ReactNode>;
+type React18ReactNode =
+  | React.ReactElement<any>
+  | string
+  | number
+  | React18ReactFragment
+  | React.ReactPortal
+  | boolean
+  | null
+  | undefined
+  | Promise<React18ReactNode>;
+
+// // React.JSXElementConstructor but it now can return React nodes from function components.
+type NewReactJSXElementConstructor<P> =
+  | ((props: P) => React18ReactNode)
+  | (new (props: P) => React.Component<P, any>);
+
+declare global {
+  namespace JSX {
+    type ElementType = string | NewReactJSXElementConstructor<any>;
+    interface IntrinsicElements {
+      ['my-web-component']: React.DOMAttributes<unknown>;
+    }
+  }
+}
+
+let Component: NewReactJSXElementConstructor<{ title: string }>;
+
+const RenderString = ({ title }: { title: string }) => title;
+Component = RenderString;
+<RenderString />;
+<RenderString title="react" />;
+<RenderString excessProp />;
+
+const RenderNumber = ({ title }: { title: string }) => title.length;
+Component = RenderNumber;
+<RenderNumber />;
+<RenderNumber title="react" />;
+<RenderNumber excessProp />;
+
+const RenderArray = ({ title }: { title: string }) => [title];
+Component = RenderArray;
+<RenderArray />;
+<RenderArray title="react" />;
+<RenderArray excessProp />;
+
+// React Server Component
+const RenderPromise = async ({ title }: { title: string }) => "react";
+Component = RenderPromise;
+<RenderPromise />;
+<RenderPromise title="react" />;
+<RenderPromise excessProp />;
+
```

Comment:
Let's make sure that class components get some level of testing here.

```suggestion

class ClassDeclaredComponent extends React.Component<{ title: string }> {
    render() {
        return <div>{this.props.title}</div>;
    }
}

<ClassDeclaredComponent />;
<ClassDeclaredComponent title="react" />;
<ClassDeclaredComponent excessProp />;

```

Reply:
done